src/m365/search/commands/externalconnection/externalconnection-add.spec.ts

```diff
@@ -0,0 +1,238 @@
+      return Promise.reject({
+        "error": {
+          "code": "Error",
+          "message": "An error has occurred",
+          "innerError": {
+            "request-id": "9b0df954-93b5-4de9-8b99-43c204a8aaf8",
+            "date": "2018-04-24T18:56:48"
+          }
+        }
+      });
+    });
+
+    command.action(logger, { options: { debug: false, subject: 'Lorem ipsum', to: 'mail@domain.com', bodyContents: 'Lorem ipsum' } } as any, (err?: any) => {
+      try {
+        assert.strictEqual(JSON.stringify(err), JSON.stringify(new CommandError(`An error has occurred`)));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('fails validation if id is not set', (done) => {
+    const actual = command.validate({
+      options: {
+        name: 'Test Connection for CLI',
+        description: 'Test connection that will not do anything'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if name is not set', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'TestConnectionForCLI',
+        description: 'Test connection that will not do anything'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if description is not set', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'TestConnectionForCLI',
+        name: 'Test Connection for CLI'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if id is less than 3 characters', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'T',
+        name: 'Test Connection for CLI'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if id is more than 32 characters', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'TestConnectionForCLIXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
+        name: 'Test Connection for CLI'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if id is not alphanumeric', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'Test_Connection!',
+        name: 'Test Connection for CLI'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if id starts with Microsoft', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'MicrosoftTestConnectionForCLI',
+        name: 'Test Connection for CLI'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
+
+  it('fails validation if id is SharePoint', (done) => {
+    const actual = command.validate({
+      options: {
+        id: 'SharePoint',
+        name: 'Test Connection for CLI'
+      }
+    });
+    assert.notStrictEqual(actual, false);
+    done();
+  });
```

Comment:
We don't need to call `done()` in these simple validation tests so they can be removed, this was also causing an error to be thrown after the tests have been executed.

```suggestion
  it('fails validation if id is SharePoint', () => {
      const actual = command.validate({
        options: {
          id: 'SharePoint',
          name: 'Test Connection for CLI'
        }
      });
      assert.notStrictEqual(actual, false);
    });
```